_posts/kevin_hartmann/2018-03-19-Drops-of-Jupyter.md

```diff
@@ -0,0 +1,145 @@
+  - *secrets*: This directory is where ansible places a copy of the SSL certifications created by certbot. If we're running in DEV_MODE, this directory can be empty.
+  - *auth_daco*: A directory that holds our python code for our DACO authentication, we have one more directory
+  - *environment*: Environment variables used to configure Jupyterhub
+  - *oath.env*: An environment variable configuration file containing our OAuth secret keys. Ansible will set this up for us when we deploy a real run; if we're running in DEV_MODE, this file can be empty.
+  - *www_custom*: The directory which stores our changes to the base jupyterhub web page look and feel. Our docker image copies this code in, over-writing existing data files in the **jupyterhub** python module 
+  - *dshell*:  A command which gives you a root shell inside the running jupyterhub docker container, so that you can make modifications to the server look and feel at run-time, and see the changes immediately, and decide which ones to keep. To make this process easier, we also mount a local directory called "share", so that we can copy web related files to/from the jupyterhub container, so it's relatively easy to make changes to the look and feel of the jupyterhub container.
+
+### Part 2: Jupyter Notebooks
+
+We have a similar set-up for the notebook environment:
+
+- Dockerfile.create-volume: Creates the full "conda" stack, and adds in any python modules that we want to mount with our /opt volume, such as our icgc-python module 
+- Dockerfile.notebook: Adds in our custom changes to the minimal Jupyterhub notebook, as well as adding in our ICGC specific code for downloading files from our cloud (ICGC-GET), and the Oracle JVM that it depends upon
+- *demo.ipynb*: The demonstration notebook that we want to add into our notebook image
+- *www_custom*: A directory containing the web related changes for to the **notebook** python module
+
+# In Summary
+<image src="{{ site.urlimg }}/kevin_hartmann/calculator.jpg" /> 
+
+We use Docker to run Jupyterhub so that we can abstract away specific details bout about how Jupyerhub is being run, like where the SSL keys are located, or which url to use for the authentication service. 
+
+That way, we can deploy Jupyterhub anywhere, just by tweaking the environment we run it in, and specifically, we can run it locally in development mode before deploying it onto our production servers. 
+
+We use ansible to take care of the details of setting up a virtual machine, setting up users and permissions, setting up environment variables with passwords and access tokens that docker will use to run, and so on... 
+
+We use multiple docker images, and a docker volume, copies the data from the docker image into the volume, run docker-compose to run jupyterhub, which launches jupyter notebooks with the volume for the conda distribution mount on /opt, which lets our users enjoy the fun of Jupyter notebooks without all of the hassle of setting up Jupyterhub. 
+
```

Comment:
In your conclusion/summary, can you add links to the actual source or more resources people could use to get started right away ?